tests/Unit/Evolution/DgSubcell/Actions/Test_Initialize.cpp

```diff
@@ -0,0 +1,301 @@
+#include "Parallel/Actions/SetupDataBox.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/AnalyticSolution.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/CloneUniquePtrs.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+struct Var1 : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+struct SystemAnalyticSolution : public MarkAsAnalyticSolution {
+  template <size_t Dim>
+  tuples::TaggedTuple<Var1> variables(
+      const tnsr::I<DataVector, Dim>& x, const double t,
+      tmpl::list<Var1> /*meta*/) const noexcept {
+    tuples::TaggedTuple<Var1> vars(x.get(0) + t);
+    for (size_t d = 1; d < Dim; ++d) {
+      get(get<Var1>(vars)) += x.get(d) + t;
+    }
+    return vars;
+  }
+
+  // clang-tidy: do not use references
+  void pup(PUP::er& /*p*/) noexcept {}  // NOLINT
+};
+
+template <size_t Dim>
+struct System {
+  static constexpr bool has_primitive_and_conservative_vars = false;
+  static constexpr size_t volume_dim = Dim;
+  using variables_tag = Tags::Variables<tmpl::list<Var1>>;
+};
+
+template <size_t Dim, typename Metavariables>
+struct Component {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using const_global_cache_tag_list = tmpl::list<>;
+
+  using initial_tags =
+      tmpl::list<Initialization::Tags::InitialTime, domain::Tags::Mesh<Dim>,
+                 domain::Tags::Element<Dim>, domain::Tags::FunctionsOfTime,
+                 domain::Tags::Coordinates<Dim, Frame::Logical>,
+                 domain::Tags::ElementMap<Dim, Frame::Grid>,
+                 domain::CoordinateMaps::Tags::CoordinateMap<Dim, Frame::Grid,
+                                                             Frame::Inertial>,
+                 Tags::Variables<tmpl::list<Var1>>>;
+
+  using phase_dependent_action_list = tmpl::list<Parallel::PhaseActions<
+      typename Metavariables::Phase, Metavariables::Phase::Initialization,
+      tmpl::list<ActionTesting::InitializeDataBox<initial_tags>,
+                 ::Actions::SetupDataBox,
+                 evolution::Initialization::Actions::SetVariables<
+                     domain::Tags::Coordinates<Dim, Frame::Logical>>,
+                 evolution::dg::subcell::Actions::Initialize<Metavariables>>>>;
+};
+
+template <size_t Dim, bool TciFails>
+struct Metavariables {
+  static constexpr size_t volume_dim = Dim;
+  using analytic_solution = SystemAnalyticSolution;
+  using component_list = tmpl::list<Component<Dim, Metavariables>>;
+  using system = System<Dim>;
+  using analytic_variables_tags = typename system::variables_tag::tags_list;
+  using const_global_cache_tags =
+      tmpl::list<Tags::AnalyticSolution<analytic_solution>>;
+  enum class Phase { Initialization, Exit };
+  struct SubcellOptions {
```

Comment:
perhaps add a pup function in anticipation of #3084

Reply:
I haven't looked through 3084, but why would we ever need to `pup` a class like this? It's never created at runtime and is just basically used as a namespace inside of `Metavariables`.

Reply:
sorry I meant add a pup to the Metavariables... (which needs to be in #3084 which always adds a DataBox tag for the metavariables, therefore requiring them to be serializable...)